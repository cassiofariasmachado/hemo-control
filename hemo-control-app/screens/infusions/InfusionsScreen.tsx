import React, { FC, useEffect, useState } from 'react';
import { InfusionsList } from '../../layouts';
import { InfusionResponse } from '../../models';
import { getInfusions } from '../../services';

type InfusionsScreenProps = { accessToken: string };

export const InfusionsScreen: FC<InfusionsScreenProps> = ({ accessToken }) => {
    const [infusions, setInfusions] = useState<InfusionResponse[]>([]);

    useEffect(() => {
        async function loadInfusions() {
            const response = await getInfusions(accessToken);

            setInfusions(response.items);
        }

        loadInfusions();
    }, [accessToken]);

    return <>
        <InfusionsList infusions={infusions} ></InfusionsList>
    </>;
};

