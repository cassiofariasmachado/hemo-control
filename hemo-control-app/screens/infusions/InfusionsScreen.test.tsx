import React from 'react';
import { create } from 'react-test-renderer';
import { InfusionsScreen } from './InfusionsScreen';

describe(InfusionsScreen.name, () => {
    test('renders correctly', () => {
        const tree = create(<InfusionsScreen accessToken={''} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
