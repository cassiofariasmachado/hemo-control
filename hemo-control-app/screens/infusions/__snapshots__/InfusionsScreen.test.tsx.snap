// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfusionsScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 50,
      },
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
